docs/iris/src/IEP/IEP001.adoc

```diff
@@ -0,0 +1,193 @@
+* It requires the use of lambda functions even when just selecting a range.
+
+Arguably, the situation when subsetting using positional indices but where the dimension order is unknown is even worse - it has no standard syntax _at all_! Instead it requires code akin to:
+[source,python]
+----
+key = [slice(None)] * cube.ndim
+key[cube.coord_dims('model_level_number')[0]] = slice(3, 9, 2)
+cube[tuple(key)]
+----
+
+The only form of indexing that is well supported is indexing by position where the dimension order is known:
+[source,python]
+----
+cube[4, 0:6, 30:]
+----
+
+## Proposal
+
+Provide indexing helpers on the Cube to extend explicit support to all permutations of:
+
+* implicit dimension vs. named coordinate,
+* and positional vs. coordinate-value based selection.
+
+### Helper syntax options
+
+Commonly, the names of coordinates are also valid Python identifiers.
+For names where this is not true, the names can expressed through either the `helper[...]` or `helper(...)` syntax by constructing an explicit dict.
+For example: `cube.loc[{'12': 0}]` or `cube.loc(**{'12': 0})`.
+
+#### Extended pandas style
+
+Use a single helper for index by position, and a single helper for index by value. Helper names taken from pandas, but their behaviour is extended by making them callable to support named coordinates.
+
+|===
+.2+| 2+h|Index by
+h|Position h|Value
+
+h|Implicit dimension
+
+a|[source,python]
+----
+cube[:, 2]  # No change
+cube.iloc[:, 2]
+----
+
+a|[source,python]
+----
+cube.loc[:, 1.5]
+----
+
+h|Coordinate name
+
+a|[source,python]
+----
+cube[dict(height=2)]
+cube.iloc[dict(height=2)]
+cube.iloc(height=2)
+----
+
+a|[source,python]
+----
+cube.loc[dict(height=1.5)]
+cube.loc(height=1.5)
+----
+|===
+
+#### xarray style
+
+xarray introduces a second set of helpers for accessing named dimensions that provide the callable syntax `(foo=...)`.
+
+|===
+.2+| 2+h|Index by
+h|Position h|Value
+
+h|Implicit dimension
+
+a|[source,python]
+----
+cube[:, 2]  # No change
+----
+
+a|[source,python]
+----
+cube.loc[:, 1.5]
+----
+
+h|Coordinate name
+
+a|[source,python]
+----
+ cube[dict(height=2)]
+ cube.isel(height=2)
+----
+
+a|[source,python]
+----
+cube.loc[dict(height=1.5)]
+cube.sel(height=1.5)
+----
+|===
+
+### Slices
+
+The semantics of position-based slices will continue to match that of normal Python slices. The start position is included, the end position is excluded.
+
+Value-based slices will be stricly inclusive, with both the start and end values included. This behaviour differs from normal Python slices but is in common with pandas.
+
+Just as for normal Python slices, we do not need to provide the ability to control the include/exclude behaviour for slicing.
+
+### Value-based indexing
+
+#### Equality
+
+Should the behaviour of value-based equality depend on the data type of the coordinate?
+
+* integer: exact match
+* float: tolerance match, tolerance determined by bit-width
```

Comment:
this always makes me a little nervous.  I can very much understand the approach tho

for interest, if I want to specify an inequality, how might i do such a thing?  does this mean that I have to deal with the situation that it may match multiple values?